examples/simplecell/simplecell.ipynb

```diff
@@ -0,0 +1,861 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import json\n",
+    "\n",
+    "from bluepyemodel.emodel_pipeline.emodel_pipeline import EModel_pipeline\n",
+    "from bluepyemodel.efeatures_extraction.targets_configurator import TargetsConfigurator"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Clear any existing checkpoints to avoid conflicts with previous runs\n",
+    "!rm -r ./checkpoints"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Setting up the pipeline"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The pipeline setup involves six key steps: **extraction** of e-features from electrophysiological recordings; **optimisation** of a NEURON cell model based on these e-features; storing the optimised model parameters; **validating** the models against specified protocols; and **plotting** the results, including traces, e-feature scores, and parameter distributions."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The [recipes.json](./config/recipes.json) file (displayed below) contains the key settings for the various stages of the e-model building pipeline.\n",
+    "\n",
+    "* **``morph_path``**: Directory path for morphologies\n",
+    "* **``morphology``**: Contains a list with the morphology's arbitrary name and file name, located in `morph_path`.\n",
+    "* **``params``**: Specifies mechanisms, locations, distributions, and parameters.\n",
+    "* **``features``**: Path to the file with extraction outputs.\n",
+    "- **`pipeline_settings`**: Sets up the pipeline with several configuration options, including:  \n",
+    "  - `extract_absolute_amplitudes`: Set to `true` for performing optimisation independently of rheobase (threshold current).  \n",
+    "  - `optimiser`: Specifies the optimisation algorithm, set to `\"SO-CMA\"` (a single-objective Covariance Matrix Adaptation algorithm).  \n",
+    "  - `max_ngen`: Defines the maximum number of generations for the optimisation process, set to `5`.  \n",
+    "  - `optimisation_params`: Additional optimisation parameters, such as `offspring_size` set to `20`, indicating the number of solutions generated per generation.  \n",
+    "  - `validation_protocols`: Lists protocols used for validation, e.g., `[\"IDrest_0.4\"]`.  \n",
+    "  - `morph_modifiers`: Set to an empty list `[]`, meaning no specific modifications to morphologies are applied by default.\n",
+    "  - `plot_currentscape`: We set this to False because we are using NEURON's built-in HH mechanism, where the currents are not defined as RANGE variables. As a result, Currentscape is unable to access and plot these currents."
```

Comment:
It would be nice to add a warning in BPEM when plot_currentscape is True, but we have hh mechanism. This can be done in another PR

Reply:
added on in the last commit, let me know if it is alright